codegen/aws/core/src/main/java/software/amazon/smithy/python/aws/codegen/AwsUserAgentIntegration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.smithy.python.aws.codegen;
+
+import java.util.Collections;
+import java.util.List;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolReference;
+import software.amazon.smithy.python.codegen.ConfigProperty;
+import software.amazon.smithy.python.codegen.integrations.PythonIntegration;
+import software.amazon.smithy.python.codegen.integrations.RuntimeClientPlugin;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+/**
+ * Adds a runtime plugin to set user agent.
+ */
+@SmithyInternalApi
+public class AwsUserAgentIntegration implements PythonIntegration {
```

Comment:
A `PythonIntegration` is sort of a plugin to the code generator.  This uses "[service provider interfaces](https://en.wikipedia.org/wiki/Service_provider_interface)" in Java.  

Basically it's [referenced later in the META-INF file below](https://github.com/SamRemis/smithy-python/pull/1/files#diff-5ee91aae5fd907b079c1a59b45ff74b8e16033ddf8b37b1badff84c7279a12a3R8) and some java magic happens that allows Smithy to pick this up during build time.  This is handled by smithy itself, not smithy-python, so we don't have to understand this one on low level just yet.